README.md

```diff
@@ -83,7 +83,7 @@ module.exports = async function(fastify, opts) {
   fastify.get(
     "/",
     {
-      onRequest: [fastify.authenticate]
+      preHandler: [fastify.authenticate]
```

Comment:
This needs to be changed.

Reply:
This is the part that must be changed (I think that for now we should replace it with `preValidation` instead of `preHandler` as it will be performed before any validation is made and will avoid any kind of security leak).

Reply:
I got it. Let me PR back this change. Also, as a fastify n00b even if I read the life cycle schema on fastify documentation and don't have a clear understanding of what happen where and why. Should have a definition on each method. Something like we can find for a React Component https://www.tutorialspoint.com/reactjs/reactjs_component_life_cycle.htm

Reply:
A PR to https://www.fastify.io/docs/latest/Lifecycle/ would be awesome!

Reply:
If I have all the info I would love to do it.